src/platform/windows/display_base.cpp

```diff
@@ -510,6 +413,22 @@ namespace platf::dxgi {
     return false;
   }
 
+  /**
+   * @brief Hook for NtGdiDdDDIGetCachedHybridQueryValue() from win32u.dll.
+   * @param gpuPreference On output, the preference value.
+   * @return S_OK always.
+   */
+  HRESULT
+  __stdcall NtGdiDdDDIGetCachedHybridQueryValueHook(D3DKMT_GPU_PREFERENCE_QUERY_STATE *gpuPreference) {
+    // By faking a cached GPU preference state of D3DKMT_GPU_PREFERENCE_STATE_UNSPECIFIED, this will
+    // prevent DXGI from performing the normal GPU preference resolution that looks at the registry,
+    // power settings, and the hybrid adapter DDI interface to pick a GPU. Instead, we will not be
+    // bound to any specific GPU. This will prevent DXGI from performing output reparenting (moving
+    // outputs from their true location to the render GPU), which breaks DDA.
+    *gpuPreference = D3DKMT_GPU_PREFERENCE_STATE_UNSPECIFIED;
```

Comment:
Should probably handle a `nullptr` somehow.

Reply:
When I test the real syscall, it returns `STATUS_INVALID_PARAMETER` in that case, so I'll do that.